src/main/scala/is/hail/variant/VariantView.scala

```diff
@@ -0,0 +1,61 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+
+class VariantView(t: TStruct) {
+  private var region: MemoryBuffer = _
+  private var offset: Long = _
+  private val contigIdx: Int = t.fieldIdx("contig")
+  private val startIdx: Int = t.fieldIdx("start")
+  private val refIdx: Int = t.fieldIdx("ref")
+  private val altAllelesIdx: Int = t.fieldIdx("altAlleles")
+  private val tAltAlleles: TArray = t.fieldType(altAllelesIdx).asInstanceOf[TArray]
+  private val tAltAllele: TStruct = tAltAlleles.elementType.asInstanceOf[TStruct]
+  private val altIdx: Int = tAltAllele.fieldIdx("alt")
+
+  // assert(t.fieldType(contigIdx).required)
```

Comment:
I don't think I can assert these right now since our representations aren't like this, but it seems like I should assert this and remove the assertions on each `get` method.